test/spec/modules/conversantAnalyticsAdapter_spec.js

```diff
@@ -0,0 +1,733 @@
+          },
+          start: 1583851418295
+        }
+      ],
+      noBids: [
+        {
+          bidder: 'appnexus',
+          params: {
+            placementId: 13144370
+          },
+          mediaTypes: {
+            banner: {
+              sizes: [
+                [
+                  300,
+                  250
+                ]
+              ]
+            }
+          },
+          adUnitCode: 'div-gpt-ad-1460505748561-0',
+          transactionId: '5fa8a7d7-2a73-4d1c-b73a-ff9f5b53ba17',
+          sizes: [
+            [
+              300,
+              250
+            ]
+          ],
+          bidId: '40a1d3ac6b79668',
+          bidderRequestId: '3e8179f67f31b98',
+          auctionId: '85e1bf44-4035-4e24-bd3c-b1ba367fe294',
+          src: 'client',
+          bidRequestsCount: 1,
+          bidderRequestsCount: 1,
+          bidderWinsCount: 0
+        }
+      ],
+      bidsReceived: [
+        {
+          bidderCode: 'conversant',
+          width: 300,
+          height: 250,
+          statusMessage: 'Bid available',
+          adId: '57e03aeafd83a68',
+          requestId: '2c2a5485a076898',
+          mediaType: 'banner',
+          source: 'client',
+          currency: 'USD',
+          cpm: 4,
+          creativeId: '29123_55016759',
+          ttl: 300,
+          netRevenue: true,
+          ad: '<foobar add goes here />',
+          originalCpm: 0.04,
+          originalCurrency: 'USD',
+          auctionId: '85e1bf44-4035-4e24-bd3c-b1ba367fe294',
+          responseTimestamp: 1583851418626,
+          requestTimestamp: 1583851418292,
+          bidder: 'conversant',
+          adUnitCode: 'div-gpt-ad-1460505748561-0',
+          timeToRespond: 334,
+          pbLg: '4.00',
+          pbMg: '4.00',
+          pbHg: '4.00',
+          pbAg: '4.00',
+          pbDg: '4.00',
+          pbCg: '',
+          size: '300x250',
+          adserverTargeting: {
+            hb_bidder: 'conversant',
+            hb_adid: '57e03aeafd83a68',
+            hb_pb: '4.00',
+            hb_size: '300x250',
+            hb_source: 'client',
+            hb_format: 'banner'
+          }
+        }
+      ],
+      winningBids: [],
+      timeout: 3000
+    };
+
+    beforeEach(function () {
+      conversantAnalytics.enableAnalytics(VALID_ALWAYS_SAMPLE_CONFIG);
+    });
+
+    afterEach(function () {
+      conversantAnalytics.disableAnalytics();
+    });
+
+    it('should not do anything when auction id doesnt exist', function() {
+      sandbox.stub(utils, 'logError');
+
+      let BAD_ARGS = JSON.parse(JSON.stringify(AUCTION_END_PAYLOAD));
+      delete BAD_ARGS.auctionId;
+      expect(requests.length).to.equal(0);
+      events.emit(constants.EVENTS.AUCTION_END, BAD_ARGS);
+      expect(requests.length).to.equal(0);
+      expect(
+        utils.logError.calledWith(
+          caObj.LOG_PREFIX + 'onAuctionEnd(): No auctionId in args supplied so unable to process event.'
+        )
+      ).to.be.true;
```

Comment:
How likely is that?  If this does happen, shouldn't we have a special event to keep track rather than tossing them?

Reply:
Pretty unlikely I'm assuming but since this is coming from an outside source that we don't control, who's to say. In any event it's an error condition that I handle so for code coverage testing it's worthwhile to test.  I don't know what we would do with that information, without an auctionId a lot of stuff falls apart, one possibility is that we send a dummy payload back with an message in the message field.

Reply:
Yeah a dummy record is what I meant by 'special event'.  Once they are sent back, at least we can write jobs to find orphan'ed records.